### PR TITLE
dotCMS/core#26231 Implement a trim input text directive

### LINKED ISSUE #26231: [UI] Implement a trim input text directive
### Parent Issue

#25716

### Problem Statement

Not allow emitting a text with white spaces at the beginning or at the end of a text input

### Steps to Reproduce

- Go to Experiment
- Select a Draft Experiment (or create a new one)
- Add a new variant
- Edit the name of the variant
- add spaces to the name at the beginning
- save the new name

### Acceptance Criteria

- Do Not allow input with spaces at the beginning or the end (Trim)
- Add to name when you add an Experiment
- Add to name when you add a Variant
- Add to name when you name a Goal
- Add to DotExperimentsInlineEditTextComponent when you edit inline a description of the Experiment or rename a Variant

### dotCMS Version

master

### Proposed Objective

User Experience

### Proposed Priority

Priority 4 - Trivial

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Do Not allow input with spaces at the beginning or the end (Trim)
* Add to `name` when you add an Experiment
* Add to `name` when you add a Variant
* Add to `name` when you name a Goal
* Add to `DotExperimentsInlineEditTextComponent` when you edit inline a `description` of the Experiment or rename a Variant

### Checklist
- [X] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### <samp>🤖 Generated by Copilot at cde0776</samp>

This pull request adds a new directive `DotExperimentsTrimInputDirective` to trim the input value on blur event for the dot-experiments portlet. It also applies this directive to several components that use input fields for experiment or variant names. Additionally, it improves the code readability, the styles, and the storybook configuration for the dot-experiments portlet. It also adds storybook stories and unit tests for the `DotExperimentsInlineEditTextComponent` and the `DotExperimentsTrimInputDirective`.

## Related Issue
Fixes #26231 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at cde0776</samp>

*  Add `DotExperimentsTrimInputDirective` to trim input values on blur event and prevent unnecessary spaces in experiment or variant name fields ([link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-7829f237bc50949e73d4183a4e4dd66a18bdb8c22b6dfa1feece1f9410556e3bR36),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-7829f237bc50949e73d4183a4e4dd66a18bdb8c22b6dfa1feece1f9410556e3bL63-R65),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8d05a2490b7df59513e786479f80e0dae9e293734e514185ceb95fd3c9f9ecbfR25),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8d05a2490b7df59513e786479f80e0dae9e293734e514185ceb95fd3c9f9ecbfL44-R46),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-c9ea183bc591a7498975f3b5812f978e7fbcade5bfc9d42f9b8ba58f0778addbR20),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-c9ea183bc591a7498975f3b5812f978e7fbcade5bfc9d42f9b8ba58f0778addbL48-R50),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-5c6947d2062e71064070479c454fcd9fe59615aa3ef621749925c7dd45bb6c06R1-R40),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-9f952f638c38547f34907d8dc16343b67a8d8be5e8553372c6fd0cbd09b15ab0R1-R19),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-712259052dae3d68ae013bbfce2f832e82c0596399a06ef2d4cc0b32135d3ac2L34-R42),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232R30-R31),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L54-R57))
*  Apply `dotTrimInput` attribute to input elements in `dot-experiments-configuration-goal-select.component.html`, `dot-experiments-configuration-variants-add.component.html`, `dot-experiments-create.component.html`, and `dot-experiments-inline-edit-text.component.html` to use the directive ([link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-a1365076c057aa6fe7e0834677535a9f19d9716de8dc4ec9d4cc9cdb58e662f6L22-R24),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8847d36f334610d5aacc1c22835a517ee9e7ad97cfdaa88fb8126b91113d3d74R25),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-712259052dae3d68ae013bbfce2f832e82c0596399a06ef2d4cc0b32135d3ac2L34-R42))
*  Add storybook stories for `DotExperimentsInlineEditTextComponent` in `dot-experiments-inline-edit-text.stories.ts` to showcase the different states and props of the component ([link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-aa34e921aeb3bcf415f301ae90b11ce585c2fa09c12d37cc14df1371f64d1441R1-R78))
*  Add default value of false to `required` input property and disable text control based on `isLoading` input property in `dot-experiments-inline-edit-text.component.ts` to handle error and loading scenarios ([link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232L101-R104),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-4665eb4639984363d439a7f16a812c9d61acdaef734c793e2dae6b6b0e5df232R141-R144))
*  Remove unnecessary selector and add missing style property in `dot-experiments-inline-edit-text.component.scss` to improve performance and consistency of input element width ([link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-f95f904546653293c6007dbd46e2c48ede229ffcb3cf210df651eecfe39848e8L20-R20),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-f95f904546653293c6007dbd46e2c48ede229ffcb3cf210df651eecfe39848e8R27))
*  Remove extra line breaks in various HTML files to improve code readability and consistency ([link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L6-R9),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L18-R16),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L27-R24),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L34-R30),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L41-R36),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L52-R46),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L69-R70),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-2d233739b679a4e33026a89a245b8a88a6b247d6eaf6e367739847208b1bcf52L95-R87),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-a1365076c057aa6fe7e0834677535a9f19d9716de8dc4ec9d4cc9cdb58e662f6L5-R5),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-a1365076c057aa6fe7e0834677535a9f19d9716de8dc4ec9d4cc9cdb58e662f6L14-R13),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-a1365076c057aa6fe7e0834677535a9f19d9716de8dc4ec9d4cc9cdb58e662f6L33-R34),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-a1365076c057aa6fe7e0834677535a9f19d9716de8dc4ec9d4cc9cdb58e662f6L52-R49),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8847d36f334610d5aacc1c22835a517ee9e7ad97cfdaa88fb8126b91113d3d74L5-R5),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8847d36f334610d5aacc1c22835a517ee9e7ad97cfdaa88fb8126b91113d3d74L15-R14),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8847d36f334610d5aacc1c22835a517ee9e7ad97cfdaa88fb8126b91113d3d74L32-R33),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8847d36f334610d5aacc1c22835a517ee9e7ad97cfdaa88fb8126b91113d3d74L51-R51),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-8847d36f334610d5aacc1c22835a517ee9e7ad97cfdaa88fb8126b91113d3d74L71-R66),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-712259052dae3d68ae013bbfce2f832e82c0596399a06ef2d4cc0b32135d3ac2L13-R13),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-712259052dae3d68ae013bbfce2f832e82c0596399a06ef2d4cc0b32135d3ac2L52-R50))
*  Add glob patterns to `stories` array in `main.js` and `include` array in `tsconfig.json` of storybook configuration to load stories and files from portlets library ([link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-c82c8893dc8aececc5879df411b5bd61f72926e31ee4202501df37b2e320cf6fL16-R17),[link](https://github.com/dotCMS/core/pull/26266/files?diff=unified&w=0#diff-d0a80a23331800c2fd194c9b9af19ba0d503f14340e5469f13c85a5c8fd0e838L12-R13))

### CODE REVIEW

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/directives/dot-experiment-trim-input/dot-experiments-trim-input.directive.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { NgControl } from '@angular/forms';
+
+/**
+ * Directive to trim input value on blur event.
+ */
+@Directive({
+    selector: '[dotTrimInput]',
+    standalone: true
+})
+export class DotExperimentsTrimInputDirective {
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/directives/dot-experiment-trim-input/dot-experiments-trim-input.directive.ts:11]: If this directive can be used in other places besides the experiment portlet, should we rename the directive to 'dotTrimInputDirective' and move it to `ui` lib?

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/dot-experiments-inline-edit-text/dot-experiments-inline-edit-text.component.ts [hunk 2]
```diff
@@ -134,6 +137,10 @@ export class DotExperimentsInlineEditTextComponent implements OnChanges {
             this.deactivateInplace();
         }
 
+        isLoading && isLoading.currentValue === true
+            ? this.textControl.disable()
+            : this.textControl.enable();
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/dot-experiments-inline-edit-text/dot-experiments-inline-edit-text.component.ts:142]: Should we test this?

**Review by rjvelazco — DISMISSED** [review]

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/directives/dot-experiment-trim-input/dot-experiments-trim-input.directive.ts (hunk 1 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/directives/dot-experiment-trim-input/dot-experiments-trim-input.directive.ts:11]: Yes. Good idea.

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/dot-experiments-inline-edit-text/dot-experiments-inline-edit-text.component.ts (hunk 2 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/dot-experiments-inline-edit-text/dot-experiments-inline-edit-text.component.ts:142]: Yes, we should.

**Review by zJaaal — DISMISSED** [review]

**Review by manuelrojas — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/dot-experiments-inline-edit-text/dot-experiments-inline-edit-text.component.ts [hunk 3]
```diff
@@ -134,6 +135,10 @@ export class DotExperimentsInlineEditTextComponent implements OnChanges {
             this.deactivateInplace();
         }
 
+        isLoading && isLoading.currentValue === true
```
Comment (fmontes) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/shared/ui/dot-experiments-inline-edit-text/dot-experiments-inline-edit-text.component.ts:138]:
```suggestion
        isLoading && isLoading.currentValue
```

File: core-web/libs/ui/src/lib/directives/dot-trim-input/dot-trim-input.directive.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        private readonly el: ElementRef
+    ) {}
+
+    @HostListener('blur')
+    onBlur() {
+        this.ngControl.control.setValue(this.ngControl.value.trim());
+    }
+
+    ngAfterViewInit(): void {
+        if (this.el.nativeElement.tagName.toLowerCase() != 'input') {
```
Comment (fmontes) [inline core-web/libs/ui/src/lib/directives/dot-trim-input/dot-trim-input.directive.ts:23]:
```suggestion
        if (this.el.nativeElement.tagName.toLowerCase() !== 'input') {
```

**Review by fmontes — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Pull request merged**